dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESMappingAPITest.java

```diff
@@ -75,11 +97,126 @@ public static void prepare() throws Exception {
+        final Map<String,Object>  contentletMap = esMappingAPI.toMap(APILocator.getContentletAPI().checkin(fileAsset, user, false));
+
+        assertNotNull(contentletMap);
+        assertEquals(fileNameField1.toLowerCase(), contentletMap.get("fileasset.filename"));
+        assertEquals("fileasset", contentletMap.get("structurename"));
+        assertEquals("text/plain; charset=iso-8859-1", contentletMap.get("metadata.contenttype"));
+        assertEquals("4", contentletMap.get("metadata.filesize"));
+        assertTrue( contentletMap.get("metadata.content").toString().contains("lol!"));
+
+    }
+
+    @Test
+    public void test_toMap_binary_field_shouldSuccess() throws Exception {
+
+        final ESMappingAPIImpl esMappingAPI    = new ESMappingAPIImpl();
+        final FieldAPI         fieldAPI        = APILocator.getContentTypeFieldAPI();
+        final Host host = APILocator.getHostAPI().findDefaultHost(user, false);
+        final String varname = "testcontenttypetwobinaryfields" + System.currentTimeMillis();
+
+        ContentType contentType = ContentTypeBuilder.builder(BaseContentType.CONTENT.immutableClass())
+                .description("Test ContentType Binary Fields")
+                .host(host.getIdentifier())
+                .name("Test ContentType Binary Fields")
+                .owner("owner")
+                .variable(varname)
+                .build();
+
+        contentType = contentTypeAPI.save(contentType);
+
+        Field textField = ImmutableTextField.builder()
+                .name("Title")
+                .variable("title")
+                .contentTypeId(contentType.id())
+                .dataType(DataTypes.TEXT)
+                .build();
+
+        textField = fieldAPI.save(textField, user);
+
+        //Creating First Binary Field.
+        Field binaryField1 = ImmutableBinaryField.builder()
+                .name("Binary 1")
+                .variable("binary1")
+                .contentTypeId(contentType.id())
+                .build();
+
+        binaryField1 = fieldAPI.save(binaryField1, user);
+
+        //Creating Second Binary Field.
+        Field binaryField2 = ImmutableBinaryField.builder()
+                .name("Binary 2")
+                .variable("binary2")
+                .indexed(true)
+                .searchable(true)
+                .contentTypeId(contentType.id())
+                .build();
+
+        binaryField2 = fieldAPI.save(binaryField2, user);
+
+        final Contentlet contentlet = new Contentlet();
+        contentlet.setContentType(contentType);
+        contentlet.setProperty("title", "binary1");
+
+        final String fileName1 = TEMP_FILE + System.currentTimeMillis();
+        final File binary1 = File.createTempFile(fileName1, TXT);
+        final String anyContent = "LOL!";
+        FileUtil.write(binary1, anyContent);
+        final File binary2 = new File(ESMappingAPITest.class.getClassLoader().getResource("images/test.jpg").getFile());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [In J2EE, getClassLoader() might not work as expected.  Use Thread.currentThread().getContextClassLoader() instead.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4960665)